test/parallel/test-diagnostics-channel-tracing-channel-promise-run-stores.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const common = require('../common');
+const { setTimeout } = require('node:timers/promises');
+const { AsyncLocalStorage } = require('async_hooks');
+const dc = require('diagnostics_channel');
+const assert = require('assert');
+
+const channel = dc.tracingChannel('test');
+const store = new AsyncLocalStorage();
+
+const context = { foo: 'bar' };
+
+channel.start.bindStore(store, common.mustCall(() => {
+  return context;
+}));
```

Comment:
Should we add something similar as in the callback case here:
```
channel.asyncStart.bindStore(store, common.mustCall(() => {
  return secondContext;
}));
```
But asserts stay as is to actually verify the runStores is not used in promise case?

Not meant as blocking comment, perfectly fine to do this in a followup or skip at all.

Reply:
I think I'll do that as a follow up. Don't want to tempt fate with CI the day before v20 cut-off. 😂